pandas/core/missing.py

```diff
@@ -248,6 +259,13 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
         except ImportError:
             raise ImportError("Your version of Scipy does not support "
                               "Akima interpolation.")
+    elif method == 'piecewise_polynomial':  # GH12887
```

Comment:
Great idea! I've incorporated your suggested convenience function. 

I'm working on a test right now that verifies that if `interpolate` is passed `method='piecewise_polynomial'`, it makes the switch and returns the expected results for `scipy > 0.17`, however I'm using a virtualenv with `pandas==0.17.0`, `scipy==0.15.1` and `numpy==1.9.2` to verify what the expected result should look like back when `piecewise_polynomial` was valid and I'm getting a `ZeroDivisionError` when passing `method='piecewise_polynomial'` to `interpolate`. 

I was looking around and found pydata/pandas#10365 which also references a couple other issues surrounding this method; pydata/pandas#10383 and pydata/pandas#10435.

So I thought I'd try `'from_derivatives'` directly, however that fails as well...
```
>>> import pandas as pd;import numpy as np
>>> s = pd.Series([0,np.nan,2])
>>> s.interpolate(method='from_derivatives')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jason/source/dev/pandas-jmwoloso/pandas/core/generic.py", line 3575, in interpolate
    **kwargs)
  File "/home/jason/source/dev/pandas-jmwoloso/pandas/core/internals.py", line 2926, in interpolate
    return self.apply('interpolate', **kwargs)
  File "/home/jason/source/dev/pandas-jmwoloso/pandas/core/internals.py", line 2890, in apply
    applied = getattr(b, f)(**kwargs)
  File "/home/jason/source/dev/pandas-jmwoloso/pandas/core/internals.py", line 906, in interpolate
    downcast=downcast, mgr=mgr, **kwargs)
  File "/home/jason/source/dev/pandas-jmwoloso/pandas/core/internals.py", line 969, in _interpolate
    interp_values = np.apply_along_axis(func, axis, data)
  File "/home/jason/.virtualenvs/pandas_dev/lib/python3.4/site-packages/numpy/lib/shape_base.py", line 115, in apply_along_axis
    outshape[axis] = len(res)
TypeError: object of type 'NoneType' has no len()
```

For dataframes, both `'piecewise_polynomial'` and `'from_derivatives'` interpolation methods just return the original dataframe without filling in `np.nan` values.